source/ddbus/meson.build

```diff
@@ -0,0 +1,37 @@
+lib_src = [
+    'attributes.d',
+    'bus.d',
+    'c_lib.d',
+    'conv.d',
+    'exception.d',
+    'package.d',
+    'router.d',
+    'simple.d',
+    'thin.d',
+    'util.d',
+]
+
+ddbus_lib = library(
+    'ddbus',
+    [lib_src],
+    install: true,
+    version: meson.project_version(),
+    soversion: project_soversion,
+    d_module_versions: d_mod_version,
+    include_directories: src_inc,
+    link_args: ['-shared', link_runtime_shared_arg],
+)
+
+pkgc = import('pkgconfig')
+
+pkgc.generate(
+    ddbus_lib,
+    name: 'ddbus-3',
```

Comment:
Why are we overwriting the name?

Reply:
Otherwise the name is `ddbus`, which would cause things to break with the next API break, so `ddbus-3` is a better idea here IMO.

Reply:
You should use soname/soversion to handle API breakages.
This is why `/usr/lib/libphobos2-ldc-shared.so.2.0.90` have the version at the end, and how `ldconfig` created the symlinks called `/usr/lib/libphobos2-ldc-shared.so.90` and `/usr/lib/libphobos2-ldc-shared.so` to the correct .so file.

Example: https://github.com/dlang-community/stdx-allocator/blob/4903a249e83b9797cc1d5167a13701a6dfd111a7/meson.build#L54

Reply:
Both pkg-config files and sonames should be used to indicate API breakages. See pipewire for a refent example of this. This is used in many places , for example glibd provides the glibd-2 pkgconfig file.

Reply:
The library still is versioned, meson uses the first digit of the version by default for the soname (so this creates libddbus.so, libddbus.so.3 and libddbus.so.3.0.0)

Reply:
But if you already have sonames, you don't need the `-3` on the name? As if the soversion is changed it mean the ABI & API changed.

Reply:
Yes, but if we install it as `ddbus` it wouldn't be possible to install different ddbus versions next to each other. This also makes it possible for applications to select what ddbus version they want by doing dependency('ddbus-2') or dependency('ddbus-3')

Reply:
That is a distro thing, not a downstream packaging thing.
If a distro wants to be able to mix version, it is their job to solve it. Like how NixOS solves it.
A downstream package should just provide the current version i.e. `ddbus`.
If a distro want to have `libddbus-old-version-99.so` it is their job to patch the build scripts.

For example, if I wanted to package two version of `ddbus` on Arch (which will never happen fyi), it is my job to make sure that both versions can be installed at the same time. It is not the job of the ddbus developers.

Reply:
That is just wrong.

```meson
dependency('ddbus', version : '>=2.0.0 <3.0.0')
```

Reply:
> That is a distro thing, not a downstream packaging thing.

It definitely is a downstream thing, on my system 58 of the pkgcofig files have a suffix with a number. Imagine the mess if distros had to coordinate the whole gtk+2/3/4 stuff because downstream didn't do that for us pkgconfig

> That is just wrong.

That may work well if you're statically linking, but this will break down once you have multiple versions around. E.g. pipewire had a 0.3 and a 0.2 pkgconfig file so distros had a smooth upgrading experience. And if gtk+ did the same thing you couldn't build gtk+2/3/4 apps without symlinking hacks. But oh well, if you don't want this I can revert it...

Reply:
> It definitely is a downstream thing, on my system 58 of the pkgcofig files have a suffix with a number. Imagine the mess if distros had to coordinate the whole gtk+2/3/4 stuff because downstream didn't do that for us pkgconfig

It's not hard... They just add the suffix...

> That may work well if you're statically linking, but this will break down once you have multiple versions around.

pkgconfig is perfectly capable to handle that.

> E.g. pipewire had a 0.3 and a 0.2 pkgconfig file so distros had a smooth upgrading experience.

That shouldn't be needed unless 0.3 is not backwards compatible with 0.2, which also requires a soversion bump. And like I said above, it is really not hard for distros to install the two versions if they want.

> And if gtk+ did the same thing you couldn't build gtk+2/3/4 apps without symlinking hacks. But oh well, if you don't want this I can revert it...

Of course you could... where is this coming from!? I don't know why GTK does it but is definitely not standard practice.

Reply:
> pkgconfig is perfectly capable to handle that.

Yes, it's perfectly capable of doing it by adding -${APIVER} as a suffix to the pkgconfig file.

> It's not hard... They just add the suffix...

If distros "just add the suffix" downstream then they'll have to either patch the build for every package that needs that package (since `dependency('ddbus')` wouldn't work then, you'd need `dependency('ddbus-3')`), or they'd need to do some symlinking action (which could be easy to mess up, accidentally providing the wrong version).

> Of course you could... where is this coming from!? I don't know why GTK does it but is definitely not standard practice.

Now I'm genuinely curious, how would packages reliably pull in the right gtk major version they need without the pkgconfig suffix? If I have gtk2/3/4 all installed on my system (all providing the same file so I'll either need something like update-alternatives on Ubuntu or have to move them to a file with a suffix myself and patch everything), how would I ensure that it picks gtk2 and not fails because it picks the wrong gtk version (and `dependency('gtk', version: '>=4')` wouldn't really work since the `gtk` pgkconfig file can be either gtk2, gtk3 or gtk4, so you'd need to symlink fron a suffixed file to gtk.

Reply:
Anyway, I've changed this to ddbus.